src/packages/CircularProgress/CircularProgress.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './circular-progress.css';
+
+const CircularProgress = ({ color, size, strokeWidth, overlay, overlayFill }) => {
+    const wrapperStyle = { width: size, height: size };
+    const spinnerLayerStyle = { borderColor: color };
+    const circleStyle = { borderWidth: strokeWidth };
+
+    const circularProgress = (
+        <div className="d2ui-circular-progress" style={wrapperStyle}>
+            <div className="spinner-layer" style={spinnerLayerStyle}>
+                <div className="circle-clipper left">
+                    <div className="circle" style={circleStyle} />
+                </div>
+                <div className="gap-patch">
+                    <div className="circle" />
+                </div>
+                <div className="circle-clipper right">
+                    <div className="circle" style={circleStyle} />
+                </div>
+            </div>
+        </div>
+    );
+
+    if (overlay) {
+        return (
+            <div className="d2ui-loader-overlay">
+                {overlayFill && (
+                    <div className="backdrop" style={{ backgroundColor: overlayFill }} />
+                )}
+                {circularProgress}
+            </div>
+        );
+    }
+
+    return circularProgress;
+};
+
+CircularProgress.propTypes = {
```

Comment:
I do not think we need the ability to customize `color`, `size`, `strokeWidth`, `overlay`, and `overlayFill` separately.

We can instead use two different classes, one for large spinners and one for small spinners.

Large spinners use the overlay to block out big chunks of content (e.g. a Table loading data, or the entire app).

A small spinner is used for inline spinning needs, e.g. inside a "save" button after it's been clicked.

Reply:
I agree but would suggest we keep some ability to either include or omit a backdrop. Not offer an option to style it, but do offer an option toggle it on or off.

Reply:
When would you want to overlay without a backdrop?

Reply:
> When would you want to overlay without a backdrop?
Some examples:
Without backdrop: for example when a form is loading and we are just showing an empty space, I would want to show the overlay spinner which is centered nicely without a backdrop
With backdrop: When a full UI is already on the page but for whatever reason you want to block a section or the whole page....

Reply:
>Without backdrop: for example when a form is loading and we are just showing an empty space, I would want to show the overlay spinner which is centered nicely without a backdrop

If we are showing an empty space it's the same with or without the backdrop, given that the backdrop color is: `rgba(255,255,255,0.7)`.

Reply:
After discussion on Slack, we decided to completely ditch the overlay and backdrop. The current components only render a ProgressIndicator. If we want them in a modal with or without backdrop this should be done via a Modal component.